3rdparty/symcrypt_engine/CMakeLists.txt

```diff
@@ -0,0 +1,127 @@
+include(FetchContent)
+FetchContent_Declare(
+  symcrypt_package
+  SOURCE_DIR ${CMAKE_CURRENT_BINARY_DIR}/SymCrypt
+  URL https://github.com/microsoft/SymCrypt/releases/download/SymCrypt_v100.20_master_2021-12-01T112555-0800_31fb2e9/symcrypt_AMD64_openenclave_100.20.0-31fb2e94-pre.tgz
+  URL_HASH
+    SHA256=8d08f459e57751cd94580395922be905f51c172dd06f3ea0db403febb7b53e3e)
+    
+# Make the downloaded package globally available
+FetchContent_GetProperties(symcrypt_package)
+if (NOT symcrypt_package_POPULATED)
+  FetchContent_Populate(symcrypt_package)
+endif ()
+
+add_enclave_library(
+  symcryptengine
+  STATIC
+  ${SYMCRYPT_ENGINE_DIR}/src/scossl.c
+  ${SYMCRYPT_ENGINE_DIR}/src/scossl_ciphers.c
+  ${SYMCRYPT_ENGINE_DIR}/src/scossl_dh.c
+  ${SYMCRYPT_ENGINE_DIR}/src/scossl_digests.c
+  ${SYMCRYPT_ENGINE_DIR}/src/scossl_dsa.c
+  ${SYMCRYPT_ENGINE_DIR}/src/scossl_ecc.c
+  ${SYMCRYPT_ENGINE_DIR}/src/scossl_helpers.c
+  ${SYMCRYPT_ENGINE_DIR}/src/scossl_hkdf.c
+  ${SYMCRYPT_ENGINE_DIR}/src/scossl_pkey_meths.c
+  ${SYMCRYPT_ENGINE_DIR}/src/scossl_rand.c
+  ${SYMCRYPT_ENGINE_DIR}/src/scossl_rsa.c
+  ${SYMCRYPT_ENGINE_DIR}/src/scossl_rsapss.c
+  ${SYMCRYPT_ENGINE_DIR}/src/scossl_tls1prf.c)
+
+enclave_compile_definitions(symcryptengine PRIVATE -D_AMD64_)
+
+if (uppercase_CMAKE_BUILD_TYPE STREQUAL "DEBUG")
+  enclave_compile_definitions(symcryptengine PRIVATE -DDBG=1)
+endif ()
+
+enclave_include_directories(
+  symcryptengine
+  PRIVATE
+  ${SYMCRYPT_ENGINE_DIR}/inc
+  ${SYMCRYPT_ENGINE_DIR}/src
+  ${CMAKE_CURRENT_BINARY_DIR}/SymCrypt/inc
+  ${CMAKE_BINARY_DIR}/3rdparty/openssl/build/include # for opensslconf.h
+  ${OPENSSL_DIR}/include)
+
+maybe_build_using_clangw(symcryptengine)
+
+if (UNIX)
+  string(REPLACE "." ";" CLANG_VERSION_LIST ${CMAKE_C_COMPILER_VERSION})
+  list(GET CLANG_VERSION_LIST 0 CLANG_MAJOR_VERSION)
+
+  set(
+    CLANG_INTRINSICS_INCLUDE_SEARCH_PATH 
+      "/usr/lib/llvm-${CLANG_MAJOR_VERSION}/lib/clang/${CMAKE_C_COMPILER_VERSION}/include"
+  )
+elseif (WIN32)
+  # Update the string when supported clang version is changed
+  set(CLANG_VERSION "10.0.0")
+
+  set(
+    CLANG_INTRINSICS_INCLUDE_SEARCH_PATH 
+      "C:/Program Files/LLVM/lib/clang/${CLANG_VERSION}/include"
+  )
+endif ()
+
+message("search path ${CLANG_INTRINSICS_INCLUDE_SEARCH_PATH}")
+
+# Clang major version should match LLVM version
+find_path(
+  CLANG_INTRINSICS_INCLUDE_DIR
+  NAMES "emmintrin.h"
+  PATHS ${CLANG_INTRINSICS_INCLUDE_SEARCH_PATH}
+  NO_DEFAULT_PATH)
+
+if (CLANG_INTRINSICS_INCLUDE_DIR)
+  enclave_compile_options(symcryptengine PRIVATE
+                          -I${CLANG_INTRINSICS_INCLUDE_DIR})
+else ()
+  message(
+    FATAL_ERROR
+      "Cannot find clang intrinstics header (required by SymCrypt OpenSSL engine)"
+  )
+endif ()
+
+enclave_compile_options(
+  symcryptengine
+  PRIVATE
+  # SymCrypt OpenSSL engine flags
+  -maes
+  -mpclmul
+  -mrdrnd
+  -mrdseed
+  -mssse3
+  -mxsave
+  # Supress warnings from SymCrypt OpenSSL engine
```

Comment:
are these OK to ignore or should they be fixed instead?

Reply:
it's ok for now IMO